lib/dao.js

```diff
@@ -1477,25 +1496,34 @@ DataAccessObject.update =
 DataAccessObject.updateAll = function (where, data, options, cb) {
   if (stillConnecting(this.getDataSource(), this, arguments)) return;
 
-  assert(arguments.length >= 2, 'At least two arguments are required');
-  if (options === undefined && cb === undefined) {
+  assert(arguments.length >= 1, 'At least one argument is required');
+
+  if (data === undefined && options === undefined && cb === undefined && arguments.length === 1) {
```

Comment:
>
> My counter-example is that when a method forwards it's arguments, perhaps
> amending some of them along the way, then it may pass undefined value.
>
I would think that would be another example that strengthens my case,
actually.

> However, in this specific case, I am ok with the solution here.
>
Cool...  I think you'd agree that this particular method *is* a special
case.... most methods don't have the possible side effect of overwriting
your entire dataset if the arguments are misinterpreted...